Model/IO/Export/VTU_XML/ParaviewVTKXMLexporter.cpp

```diff
@@ -0,0 +1,147 @@
+#include "ParaviewVTKXMLexporter.h"
+#include <iostream>
+
+
+namespace MeshIS
+{
+	namespace Model
+	{
+
+		using Common::Element_T4;
+		using Common::Element_P6;
+
+		namespace Export
+		{
+
+			void ParaviewVTKXMLExporter::Export(const std::string& absolute_file_path, const CMR& mesh_data)
+			{
+				checkFileToSave(absolute_file_path);
+				fileStream.open(absolute_file_path, std::ios::out | std::ios::trunc);
+				
+				if(!fileStream.good()){
+					std::cout << "Error while opening file.\nEnter will close program." << std::endl;
+					std::cin.get();
+					fileStream.close();
+					exit(0);
+				}
+				const unsigned int noOfCells = mesh_data.elementsT4.size() + mesh_data.elementsP6.size();
+
+				fileStream << "<VTKFile type=\"UnstructuredGrid\" version=\"1.0\" byte_order=\"LittleEndian\" header_type=\"UInt64\">" << std::endl;
+				fileStream << "<UnstructuredGrid>" << std::endl;
+				fileStream << "<Piece NumberOfPoints=\"" << mesh_data.vertices.size() << "\" NumberOfCells=\"" << noOfCells << "\">" << std::endl;
+				fileStream << "<PointData>" << std::endl;
+				fileStream << "</PointData>" << std::endl;
+				fileStream << "<CellData>" << std::endl;
+				fileStream << "</CellData>" << std::endl;
+				fileStream << "<Points>" << std::endl;
+				fileStream << "<DataArray type=\"Float64\" Name=\"Points\" NumberOfComponents=\"3\" format=\"ascii\">" << std::endl;
+
+				putVerticesTofile(mesh_data);
+
+				fileStream << std::endl << "</DataArray>" << std::endl;
+				fileStream << "</Points>" << std::endl;
+				fileStream << "<Cells>" << std::endl;
+				fileStream << "<DataArray type=\"Int64\" Name=\"connectivity\" format=\"ascii\">" << std::endl;
+
+				putConnectivityTofile(mesh_data);
+
+				fileStream << "</DataArray>" << std::endl;
+				fileStream << "<DataArray type=\"UInt8\" Name=\"types\" format=\"ascii\">" << std::endl;
+				
+				putTypesTofile(mesh_data);
+
+				fileStream << std::endl << "</DataArray>" << std::endl;
+				fileStream << "</Cells>" << std::endl;
+				fileStream << "</Piece>" << std::endl;
+				fileStream << "</UnstructuredGrid>" << std::endl;
+				fileStream << "</VTKFile>" << std::endl;
+				
+				fileStream.close();
+
+				std::cout << "Export Completed!" << std::endl;
+
+			}
+			void ParaviewVTKXMLExporter::checkFileToSave(const string& absolute_file_path) {	
+				const char extension[4] = { absolute_file_path[absolute_file_path.size() - 4],
```

Comment:
Did you consider using substring or other more compact method to retrive extension?